packages/eslint-plugin/src/rules/unbound-method.ts

```diff
@@ -209,6 +214,42 @@ export default createRule<Options, MessageIds>({
               );
             }
           });
+          return;
+        }
+
+        if (node.typeAnnotation) {
+          const subtypes = tsutils.unionTypeParts(
+            services.getTypeAtLocation(node.typeAnnotation.typeAnnotation),
+          );
+          for (const property of node.properties) {
+            for (const subtype of subtypes) {
+              if (
+                property.type !== AST_NODE_TYPES.Property ||
+                property.key.type !== AST_NODE_TYPES.Identifier
+              ) {
+                continue;
+              }
+
+              const propertyKey = property.key;
+              if (
+                checkIfMethodAndReport(
```

Comment:
(style) Consider assigning the result of `checkIfMethodAndReport` to a variable to indicate clearly that it has side effects.